Sources/Display.php

```diff
@@ -851,26 +852,26 @@ function Display()
 		if (isset($_SESSION['page_next_start']) && $_SESSION['page_next_start'] == $start)
 		{
 			$start_char = 'M';
-			$page_id = $_SESSION['page_last_id'];
+			$page_id = $ascending ? $_SESSION['page_last_id'] : $_SESSION['page_first_id'];
 		}
 		// User moved to the previous page
 		elseif (isset($_SESSION['page_before_start']) && $_SESSION['page_before_start'] == $start)
 		{
 			$start_char = 'L';
-			$page_id = $_SESSION['page_first_id'];
+			$page_id = $ascending ? $_SESSION['page_first_id'] : $_SESSION['page_last_id'];
 		}
 		// User refreshed the current page
 		elseif (isset($_SESSION['page_current_start']) && $_SESSION['page_current_start'] == $start)
 		{
 			$start_char = 'C';
-			$page_id = $_SESSION['page_first_id'];
+			$page_id = $ascending ? $context['topicinfo']['id_first_msg'] : $context['topicinfo']['id_last_msg'];
```

Comment:
Also, why aren't you using the `$_SESSION['page_first_id']` and `$_SESSION['page_last_id']` values in this case?

Reply:
In the first version of the pr i did this but,
then this issue: https://github.com/SimpleMachines/SMF2.1/issues/4520#issuecomment-361322540 was raised up.
in short the page_last_id is old and i needed the newest value.
This limitation was already "known" by me, reason why i didn't implement the last page.
but know i was forced to look more deeply.

Reply:
I see. If you keep track of the value of `$context['total_visible_posts']` in `$_SESSION['page_total_visible_posts']`, then you could check to see if the number of visible posts in the topic has changed since the last page load. If it has changed, then you'll know that the saved `$_SESSION['page_*']` values are no longer correct.